qrcode/main.py

```diff
@@ -23,11 +23,22 @@ def _check_box_size(size):
             "Invalid box size (was %s, expected larger than 0)" % size)
 
 
+def _check_mask_pattern(mask_pattern):
+    if mask_pattern == None:
+        return
+    if not isinstance(mask_pattern, int):
+        raise TypeError(
+            "Invalid mask pattern (was %s, expected int)" % type(mask_pattern))
+    if mask_pattern not in range(8):
+        raise ValueError(
+            "Mask pattern should be in range(8) (got %s)" % mask_pattern)
+
 class QRCode:
 
     def __init__(self, version=None,
                  error_correction=constants.ERROR_CORRECT_M,
                  box_size=10, border=4,
+                 mask_pattern=None,
```

Comment:
Best to add new arguments to the end so it won't break if anyone is only using positional arguments.